Enterprise-CMCS/macpro-mdct-mfp
# FEEDBACK REQUESTED: demo new error pattern

### Description

I noticed a bug where nested error throws were catching back at the handler-lib root catch.
Ex:
```js
// handler-lib.ts
try {
  create();
} catch (e) {
  console.error("handler caught");
  return response(500, e.message)
}

// create.ts
try {
  validateData(data);
} catch {
  console.error("create caught");
  return response(400, "invalid data")
}

// validateData.ts
try {
  data.validate();
} catch {
  console.error("validate caught");
  throw new Error("could not validate");
}

// OUTPUT
// validate caught
// handler caught
// Response: 500, could not validate
```

Although I am still baffled by this, it made me investigate how we are handling errors and if there is a cleaner way we can handle them.

Focus on the changes in `handler-lib.ts`, `reports/create.ts`, and `utils/constants/constants.ts`

This change shifts from determining status code alongside error to mapping error to status code and having our global handler lookup the appropriate status code.

I don't love this pattern. However, given the described bug, I am looking for a reasonable solution. Since our global handler will catch any error and default to 500, why not let it catch any error and use the correct status code too.

Also open to just fixing the bug if people know how!

### Related ticket(s)


started with CMDCT-3988 

---
### How to test

- Log in as a state user
- Open network tab and create a WP
- resend the WP request, changing a body type to be invalid (number to array `[]`, for example)
- verify it returns a 400 with the message "Provided data is not valid."

Compare to main where it returns a 500 with the same message

### Notes


A number of tests expected 500 status codes for things like invalid or missing data. This means we were accommodating the bug! We should be vigilant about creating tests that expect the _desired_ outcome, not the actual one.

---
### Pre-review checklist


---
### Pre-merge checklist


#### Review
---
